packages/core/src/compiler.ts

```diff
@@ -266,7 +266,15 @@ export class Compiler {
       usedFields: [],
     };
 
-    let code = transformCardComponent(templateSource, options);
+    let code: string;
+    try {
+      code = transformCardComponent(templateSource, options);
+    } catch (error: any) {
+      let errorMessage = `Failed to compile \`${localFile}\` in \`${cardURL}\``;
+      console.error(`----------------------\n${errorMessage}\n----------------------`);
+      console.error(error);
+      code = `throw new Error('${errorMessage}. Check your hub log')`;
```

Comment:
@ef4 I'm struggling to think of a way to let the error pass all the way through while watching for changes and to blow up when doing a command like priming the cache. There seems to be a conflict here. 

Some ideas I have are:
1. Keep track of this error somehow and check for it in the CLI commands, then exit
2. Don't pass along the error inside the specific module but do something similar to ember-cli where it changes the whole page to an error